### PR TITLE
Updated Java configuration with new settings

### PR DESCRIPTION
**Pull request by andrewsouthard1**
### What does this PR do?

Adds missing configuration options to Java APM documentation.

### Motivation

New changes in Java tracer that were not automatically added to docs. 

### Preview link

https://docs-staging.datadoghq.com/andrew/add-config-java-http-status/tracing/setup/java/

### CODE REVIEW

File: content/en/tracing/setup/java.md [hunk 1]
```diff
@@ -92,6 +92,11 @@ Beta integrations are disabled by default but can be enabled individually.
 *Note:* Many application servers are Servlet compatible and are automatically covered by that instrumentation, such as Tomcat, Jetty, Websphere, Weblogic, etc.
 Also, frameworks like Spring Boot inherently work because it uses a Servlet compatible embedded application server.
 
+Grizzly Instrumentation is disabled by default - to enable, add one of the following configs.
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
Grizzly Instrumentation is disabled by default. Add one of the following configs to enable it:
```

File: content/en/tracing/setup/java.md [hunk 2]
```diff
@@ -182,7 +187,7 @@ The tracer is configured using System Properties and Environment Variables as fo
...
 | `dd.trace.enabled`                     | `DD_TRACE_ENABLED`                     | `true`               | When `false` tracing agent is disabled.                                                                                                                                                                                 |
-| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. E.g. `dd.trace.enabled=my_service-name`                                                                                           |
+| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. Ex. the file path can be provided as via `-Dtrace.config=/path/to/file.properties`, with setting the service name in the file with `dd.trace.enabled=my_service-name`                                                                                           |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. For instance, the file path can be provided as via `-Dtrace.config=/path/to/file.properties`, with setting the service name in the file with `dd.trace.enabled=<SERVICE_NAME>`                                                                                           |
```

File: content/en/tracing/setup/java.md [hunk 3]
```diff
@@ -199,11 +204,11 @@ The tracer is configured using System Properties and Environment Variables as fo
...
-| `dd.jmxfetch.config.dir`          | `DD_JMXFETCH_CONFIG_DIR`          | `null`               | (Example: `/opt/datadog-agent/etc/conf.d`) Additional configuration directory for JMX metrics collection. Available in Java Datadog Trace Agent v0.29.0+.                                                                                                  |
-| `dd.jmxfetch.config`          | `DD_JMXFETCH_CONFIG`          | `null`               | (Example: `activemq.d/conf.yaml,jmx.d/conf.yaml`) Additional metrics configuration file for JMX metrics collection. Available in Java Datadog Trace Agent v0.29.0+.                                                                                                           |
+| `dd.jmxfetch.config.dir`          | `DD_JMXFETCH_CONFIG_DIR`          | `null`               | (Example: `/opt/datadog-agent/etc/conf.d`) Additional configuration directory for JMX metrics collection. The Java Agent will look for `jvm_direct:true` in the `instance` in the `yaml` file to change configuration.                                                                                                    |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.jmxfetch.config.dir`          | `DD_JMXFETCH_CONFIG_DIR`          | `null`               | (Example: `/opt/datadog-agent/etc/conf.d`) Additional configuration directory for JMX metrics collection. The Java Agent looks for `jvm_direct:true` in the `instance` section in the `yaml` file to change configuration.                                                                                                    |
```

File: content/en/tracing/setup/java.md [hunk 4]
```diff
@@ -199,11 +204,11 @@ The tracer is configured using System Properties and Environment Variables as fo
...
+| `dd.jmxfetch.config`          | `DD_JMXFETCH_CONFIG`          | `null`               | (Example: `activemq.d/conf.yaml,jmx.d/conf.yaml`) Additional metrics configuration file for JMX metrics collection. The Java Agent will look for `jvm_direct:true` in the `instance` in the `yaml` file to change configuration.                                                                                                          |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.jmxfetch.config`          | `DD_JMXFETCH_CONFIG`          | `null`               | (Example: `activemq.d/conf.yaml,jmx.d/conf.yaml`) Additional metrics configuration file for JMX metrics collection. The Java Agent looks for `jvm_direct:true` in the `instance` section in the `yaml` file to change configuration.                                                                                                          |
```

File: content/en/tracing/setup/java.md [hunk 5]
```diff
@@ -214,6 +219,9 @@ The tracer is configured using System Properties and Environment Variables as fo
 
 * If the same key type is set for both, the system property configuration takes priority.
 * System properties can be used as JVM parameters.
+* By default, JMX metrics from your application are sent to the Datadog Agent thanks to DogStatsD over port `8125`. Make sure that [DogStatsD is enabled for the Agent][12].
+If you are running the Agent as a container, ensure that `DD_DOGSTATSD_NON_LOCAL_TRAFFIC` [is set to true][13], and that port `8125` is open on the Agent.
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
If you are running the Agent as a container, ensure that `DD_DOGSTATSD_NON_LOCAL_TRAFFIC` [is set to `true`][13], and that port `8125` is open on the Agent container.
```

File: content/en/tracing/setup/java.md [hunk 6]
```diff
@@ -214,6 +219,9 @@ The tracer is configured using System Properties and Environment Variables as fo
...
+In Kubernetes, [bind the DogstatsD port to a host port][14]; in ECS, [set the appropriate flags in your task definition][15].
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
In Kubernetes, [bind the DogStatsD port to a host port][14]; in ECS, [set the appropriate flags in your task definition][15].
```

File: content/en/tracing/setup/java.md [hunk 7]
```diff
@@ -285,6 +293,8 @@ compile group: 'com.datadoghq', name: 'dd-trace-api', version: {version}
 
 Now add `@Trace` to methods to have them be traced when running with `dd-java-agent.jar`.  If the Agent is not attached, this annotation has no effect on your application.
 
+`@Trace` annotations will have the default operation name `trace.annotation`, while the method traced will by default be the resource.
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
`@Trace` annotations have the default operation name `trace.annotation`, while the method traced have the resource by default.
```

**Comment by andrewsouthard1** [comment]
Merged all changes, thanks!

File: content/en/tracing/setup/java.md [hunk 8]
```diff
@@ -92,6 +92,11 @@ Beta integrations are disabled by default but can be enabled individually.
 *Note:* Many application servers are Servlet compatible and are automatically covered by that instrumentation, such as Tomcat, Jetty, Websphere, Weblogic, etc.
 Also, frameworks like Spring Boot inherently work because it uses a Servlet compatible embedded application server.
 
+Grizzly Instrumentation is disabled by default. Add one of the following configs to enable it:
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
Grizzly Instrumentation is disabled by default. Add one of the following configurations to enable it:
```

File: content/en/tracing/setup/java.md [hunk 9]
```diff
@@ -182,7 +187,7 @@ The tracer is configured using System Properties and Environment Variables as fo
...
 | `dd.trace.enabled`                     | `DD_TRACE_ENABLED`                     | `true`               | When `false` tracing agent is disabled.                                                                                                                                                                                 |
-| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. E.g. `dd.trace.enabled=my_service-name`                                                                                           |
+| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. For instance, the file path can be provided as via `-Dtrace.config=/path/to/file.properties`, with setting the service name in the file with `dd.trace.enabled=<SERVICE_NAME>`                                                                                           |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.trace.config`                      | `DD_TRACE_CONFIG`                      | `null`               | Optional path to a file were configuration properties are provided one per each line. For instance, the file path can be provided as via `-Dtrace.config=<FILE_PATH>.properties`, with setting the service name in the file with `dd.trace.enabled=<SERVICE_NAME>`                                                                                           |
```

File: content/en/tracing/setup/java.md [hunk 10]
```diff
@@ -199,11 +204,11 @@ The tracer is configured using System Properties and Environment Variables as fo
...
 | `dd.trace.db.client.split-by-instance` | `DD_TRACE_DB_CLIENT_SPLIT_BY_INSTANCE` | `false`              | When set to `true` db spans get assigned the instance name as the service name                                                                                                                                          |
 | `dd.http.client.tag.query-string`      | `DD_HTTP_CLIENT_TAG_QUERY_STRING`      | `false`              | When set to `true` query string parameters and fragment get added to web client spans                                                                                                                                   |
+| `dd.http.client.error.statuses`      | `DD_HTTP_CLIENT_ERROR_STATUSES`      | `false`              | A range of errors can be accepted, by default 4xx errors will not be reported as errors. This configuration overrides that. Ex. `dd.http.client.error.statuses=400-499`                                                                                                                                   |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.http.client.error.statuses`      | `DD_HTTP_CLIENT_ERROR_STATUSES`      | `false`              | A range of errors can be accepted, by default 4xx errors are reported as errors. This configuration overrides that. Ex. `dd.http.client.error.statuses=400-499`                                                                                                                                   |
```

File: content/en/tracing/setup/java.md [hunk 11]
```diff
@@ -199,11 +204,11 @@ The tracer is configured using System Properties and Environment Variables as fo
...
 | `dd.trace.db.client.split-by-instance` | `DD_TRACE_DB_CLIENT_SPLIT_BY_INSTANCE` | `false`              | When set to `true` db spans get assigned the instance name as the service name                                                                                                                                          |
 | `dd.http.client.tag.query-string`      | `DD_HTTP_CLIENT_TAG_QUERY_STRING`      | `false`              | When set to `true` query string parameters and fragment get added to web client spans                                                                                                                                   |
+| `dd.http.client.error.statuses`      | `DD_HTTP_CLIENT_ERROR_STATUSES`      | `false`              | A range of errors can be accepted, by default 4xx errors are reported as errors. This configuration overrides that. Ex. `dd.http.client.error.statuses=400-499`                                                                                                                                   |
```
Comment (l0k0ms) [inline content/en/tracing/setup/java.md]:
```suggestion
| `dd.http.client.error.statuses`      | `DD_HTTP_CLIENT_ERROR_STATUSES`      | `false`              | A range of errors can be accepted, by default 4xx errors aren't reported as errors. This configuration overrides that. Ex. `dd.http.client.error.statuses=400-499`                                                                                                                                   |
```

**Pull request merged**